### PR TITLE
[docs] Add recipe for single-click editing

### LINKED ISSUE #2186: [DataGrid] Add demo for single click edit
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

Add demo for single-click edit.

https://material-ui.com/components/data-grid/editing/.

## Examples 🌈

https://codesandbox.io/s/material-demo-forked-3s7y6?file=/demo.tsx

```jsx
export default function StartEditButtonGrid() {
  const apiRef = useGridApiRef();

  const handleCellClick = React.useCallback(
    (params: GridCellParams) => {
      apiRef.current.setCellMode(params.id, params.field, "edit");
    },
    [apiRef]
  );

  return (
    <div style={{ height: 400, width: "100%" }}>
      <XGrid
        rows={rows}
        columns={columns}
        apiRef={apiRef}
        onCellClick={handleCellClick}
      />
    </div>
  );
}
```

## Motivation 🔦

This is a popular configuration, from one of our users:

> Can we make all the fields in the table always editable? Right now, we have to double-click into a field before we can edit it, which is not the ideal user experience.

or https://www.ag-grid.com/react-grid/cell-editing/#single-click-editing

### PR DESCRIPTION
**Pull request by m4theushw**
Closes #2186

### CODE REVIEW

**Comment by DannyMeister** [comment]
The code sample in this recipe suffers from a bug where the initial value is lost if the cell edit is initiated from an api call to startCellEditMode() which passes an initialValue property.  I have linked a sample fix to https://github.com/mui/mui-x/issues/2186#issuecomment-1544769292 .  For convenience here's the direct link as well. https://codesandbox.io/s/material-demo-forked-c0ldcq?file=/demo.tsx

**Comment by noah-lee** [comment]
Thank you for looking into adding a demo for this! I have been using this workaround but found a small issue when upgrading to v6 with single select edit cells:

Once the select cell menu is open, clicking away does not close/change back to view mode Reproduction: https://codesandbox.io/s/material-demo-forked-c0ldcq?file=/demo.tsx

It seems like the click event on the backdrop is passing through and triggering `handleCellClick` instead. The workaround I found was to add:

```
slotProps={{
  baseSelect: {
    onClick: (event) => event.stopPropagation()
  }
}}
```

File: docs/data/data-grid/editing/editing.md [hunk 1]
```diff
@@ -39,6 +39,10 @@ Users can start editing a cell (or row if `editMode="row"`) with any of the foll
   apiRef.current.startRowEditMode({ id: 1 });
   ```
 
+:::info
+You can also enter edit mode with a single-click by following [this recipe](/x/react-data-grid/recipes-editing/#single-click-editing).
```
Comment (cherniavskii) [inline docs/data/data-grid/editing/editing.md:43]:
```suggestion
You can also enter edit mode with a single click by following [this recipe](/x/react-data-grid/recipes-editing/#single-click-editing).
```

File: docs/data/data-grid/recipes-editing/recipes-editing.md [hunk 2]
```diff
@@ -98,6 +98,14 @@ Note that the `onCellEditStart` and `onCellEditStop` props also have to be used
 
 {{"demo": "LinkedFieldsCellEditing.js", "bg": "inline", "defaultCodeOpen": false}}
 
+## Single-click editing
+
+One of the [built-in ways](/x/react-data-grid/editing/#start-editing) to the enter mode is by double-clicking a cell.
+Using the [controlled mode](/x/react-data-grid/editing/#controlled-mode) and listening to click events, you can also enter the edit mode with just a single-click.
```
Comment (cherniavskii) [inline docs/data/data-grid/recipes-editing/recipes-editing.md:104]:
```suggestion
By default, one of the ways to [enter the edit mode](/x/react-data-grid/editing/#start-editing) is by double-clicking a cell.
Using the [controlled mode](/x/react-data-grid/editing/#controlled-mode) and listening to click events, you can also enter the edit mode with just a single click.
```

**Review by cherniavskii — APPROVED** [review]

**Pull request merged**